flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java

```diff
@@ -273,7 +276,7 @@ public void releaseOutputs() {
 
 		// now create the operator and give it the output collector to write its output to
 		OneInputStreamOperator<IN, OUT> chainedOperator = operatorConfig.getStreamOperator(userCodeClassloader);
-		chainedOperator.setup(containingTask, operatorConfig, output);
+		chainedOperator.setup(containingTask, operatorConfig, output, streamOutputs.size() == 0);
```

Comment:
The reason why we still differentiate between those is that we don't want to keep the latency statistics for all parallel source instances at the intermediate operators.
So in the current implementation:
- Intermediate operators maintain latency statistics for each logical source (so not much data)
- Sinks maintain latency statistics for all parallel instances of each logical source (a bit more data)
The sinks maintain the additional data to allow users debugging latency issues with individual parallel instances (for example when a machine running a source has issues).